plainly-plugin/src/ui/hooks/useSettings.ts

```diff
@@ -86,8 +97,48 @@ export const useSettings = () => {
     [],
   );
 
-  // TODO: handle decrypt
-  const getSettingsApiKey = () => settings.apiKey;
+  const getSettingsApiKey = useCallback(
+    (pin: string | undefined = undefined): string => {
+      const { key, encrypted } = settings.apiKey ?? {};
+
+      if (!key) {
+        throw new Error('API key is not set.');
+      }
+
+      if (pin && !encrypted) {
+        throw new Error('API key is not encrypted, pin is not required.');
+      }
+
+      if (!pin && encrypted) {
+        throw new Error('API key is encrypted, pin is required.');
+      }
+
+      if (pin && encrypted) {
+        try {
+          const decoded = decode(pin, key);
+          return decoded;
+        } catch (error) {
+          throw new Error('Invalid PIN entered.');
+        }
+      }
+
+      return key;
```

Comment:
can we please have return in `else` here so difficult to follow
```suggestion
      if (pin && encrypted) {
        try {
          const decoded = decode(pin, key);
          return decoded;
        } catch (error) {
          throw new Error('Invalid PIN entered.');
        }
      } else {
         return key;
      }
```